packages/contracts-bedrock/test/libraries/Blueprint.t.sol

```diff
@@ -4,14 +4,100 @@ pragma solidity 0.8.15;
 import { Test } from "forge-std/Test.sol";
 import { Blueprint } from "src/libraries/Blueprint.sol";
 
+// Used to test that constructor args are appended properly when deploying from a blueprint.
+contract ConstructorArgMock {
+    uint256 public x;
+    bytes public y;
+
+    constructor(uint256 _x, bytes memory _y) {
+        x = _x;
+        y = _y;
+    }
+}
+
+// Foundry cheatcodes operate on the next call, and since all library methods are internal we would
+// just JUMP to them if called directly in the test. Therefore we wrap the library in a contract.
+contract BlueprintHarness {
+    function blueprintDeployerBytecode(bytes memory _initcode) public pure returns (bytes memory) {
+        return Blueprint.blueprintDeployerBytecode(_initcode);
+    }
+
+    function parseBlueprintPreamble(bytes memory _bytecode) public pure returns (Blueprint.Preamble memory) {
+        return Blueprint.parseBlueprintPreamble(_bytecode);
+    }
+
+    function deployFrom(address _blueprint, bytes32 _salt) public returns (address) {
+        return Blueprint.deployFrom(_blueprint, _salt);
+    }
+
+    function deployFrom(address _blueprint, bytes32 _salt, bytes memory _args) public returns (address) {
+        return Blueprint.deployFrom(_blueprint, _salt, _args);
+    }
+
+    function bytesToUint(bytes memory _bytes) public pure returns (uint256) {
+        return Blueprint.bytesToUint(_bytes);
+    }
```

Comment:
I was thinking to reduce the amount of times you access assembly in the tests to perform a `create2`, you could add this function to the harness: 

```solidity
    function deployUsingCreate2(bytes memory _bytecode, bytes32 _salt) public returns (address blueprintAddress) {
        assembly ("memory-safe") {
            blueprintAddress := create2(0, add(_bytecode, 0x20), mload(_bytecode), _salt)
        }
        require(blueprintAddress != address(0), "Deploy using create2 failed");
    }
```

The use this in your tests.

Reply:
Done in 84c393a7e